core/src/mfd.rs

```diff
@@ -0,0 +1,74 @@
+//! Implementation of a memory fd
+
+use std::io::{Read, Seek, SeekFrom, Write};
+use std::os::fd::{AsRawFd, RawFd};
+
+use anyhow::{bail, Result};
+use memfd::{FileSeal, Memfd, MemfdOptions};
+
+pub struct Mfd(Memfd);
+
+impl Mfd {
+    /// Creates an empty named Memfd
+    pub fn create(name: &str) -> Result<Self> {
```

Comment:
The name is purely for debugging purposes, correct?

Reply:
It is actually part of the `memfd` interface, if you take a look at `memfd_create(2)`.

The man page says the following:
> The displayed name is always prefixed  with  memfd:  and  serves only  for  debugging purposes.  Names do not affect the behavior of the file descriptor, and as such multiple files can have the same name without any side effects.

So yes. It is indeed only for debugging